streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -459,7 +471,9 @@ void commit(final boolean startNewTransaction) {
         for (final Map.Entry<TopicPartition, Long> entry : consumedOffsets.entrySet()) {
             final TopicPartition partition = entry.getKey();
             final long offset = entry.getValue() + 1;
-            consumedOffsetsAndMetadata.put(partition, new OffsetAndMetadata(offset));
+            final long partitionTime = getNextPartitionTime(partition);
```

Comment:
I don't understand the purpose of this method. Why not just get the `partitionTime` of the task and commit it?

Reply:
Ah, this was something I added after I discovered a bug during integration tests. What happened was that offsets are periodically right (that is, it is automated)?  So imagine this, the partitionTime has advanced to 10 milliseconds. We commit that time, and then streams experienced some sort of failure (akin to a restart of streams). That would mean the locally stored partitionTime was reset to -1. When the timestamp is committed again, the correct timestamp should be  10, but instead -1 was committed. So what we need to do is retrieve the previously committed timestamp if there was any, and then commit _that_ one instead, since that is the correct one.

I had some second thoughts on this, particularly since it becomes a little difficult to distinguish between restarts, cleanups, or failures which could have the same effect. So 
I'm not so sure if this is still needed or we still need to modify the behavior for these cases.

Reply:
Actually, on further investigation, this method might not be needed. We will see.

Reply:
Ok, I realized what is happening. During the process to close a StreamTask, the partitionTimes are reset to -1 first before the local partition times are committed. Effectively, what is occurring is that we are committing -1 during close() due to the order of operations we are performing it. I have found a solution to it, so I will push a change shortly.

Reply:
Good find!